src/py-opentimelineio/opentimelineio/adapters/adapter.py

```diff
@@ -175,21 +175,30 @@ def write_to_file(
         input_otio = hooks.run("pre_adapter_write", input_otio,
                                extra_args=hook_function_argument_map)
 
+        # Store filepath for use in post_adapter_write hooks
+        hook_function_argument_map['_filepath'] = filepath
```

Comment:
you know, this is probably something we should pass to the `pre_adapter_write` function as well.  that's a good call. that way you can write a hook that is specific to a file extension or path or something.

Reply:
Moved the assignment up a few lines to make this happen.

Reply:
@ssteinbach 
It just struck me. Is the `_filepath` key name a good one? Will it make sense if/when persisted in a  "pre adapter write" hook's metadata?

Any suggestions for a better key name?

Reply:
Any thoughts on this?

Reply:
We could deep copy the `hook_function_argument_map` on entry to the function.  I'm not sure what you mean by in_timeline metadata" though, sorry.  Can you clarify?

Reply:
Sorry, meant `input_otio`'s metadata. 
Aaand I think I might have had a little brain fart here.. For some reason I was under the impression that all hooks modify the `input_otio`'s metadata based on the `hook_function_argument_map`. I think a deep copy will indeed do the trick and if a users hook script modifies the metadata including all of the `hook_function_argument_map` it'll be up to them to deal with it I guess